include/zephyr/dt-bindings/pinctrl/quicklogic-eos-s3-pinctrl.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_QUICKLOGIC_EOS_S3_PINCTRL_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_QUICKLOGIC_EOS_S3_PINCTRL_H_
+
+#include <zephyr/sys/util_macro.h>
+
+#define IO_MUX_REG_MAX_OFFSET	107
+#define IO_MUX_MAX_PAD_NR	45
+
+#define QUICKLOGIC_EOS_S3_UART_TXD_SEL_PAD44	(0x00000003)
+#define QUICKLOGIC_EOS_S3_UART_RXD_SEL_PAD45	BIT(2)
+
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_PU_CTRL_PAD	23
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_DN_PAD		28
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_DP_PAD		31
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_TX_PAD		44
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_RX_PAD		45
+
+#define QUICKLOGIC_EOS_S3_PINMUX_FUNC_SEL_UART_RX	(77 << 13)
+
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_RX_PAD_CFG	(QUICKLOGIC_EOS_S3_PINMUX_FUNC_SEL_UART_RX \
+							 | QUICKLOGIC_EOS_S3_UART_RXD_SEL_PAD45)
```

Comment:
is selection related to pad?

why not something like `#define UART_TX_PAD44 QUICKLOGIC_EOS_S3_PINMUX(44, 0x3)`

Reply:
`*_FUNC_SEL*` is related to the function and the function is `UART_rxd` in this case.
I've implemented the macro you suggested and reworked the driver. I think the DTS looks much simpler now. Thank you.